src/zenml/metadata_stores/mysql_metadata_store.py

```diff
@@ -37,11 +43,106 @@ def get_tfx_metadata_config(
         metadata_store_pb2.ConnectionConfig,
         metadata_store_pb2.MetadataStoreClientConfig,
     ]:
-        """Return tfx metadata config for mysql metadata store."""
-        return metadata.mysql_metadata_connection_config(
+        """Return tfx metadata config for MySQL metadata store."""
+        config = MySQLDatabaseConfig(
             host=self.host,
             port=self.port,
             database=self.database,
-            username=self.username,
-            password=self.password,
         )
+
+        secret = self._get_mysql_secret()
+
+        # Set the user
+        if self.username:
+            if secret and secret.user:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a username `{self.username}` and `{secret.user}`. Please "
+                    f"make sure that you only use one."
+                )
+            else:
+                config.user = self.username
+        else:
+            if secret and secret.user:
+                config.user = self.username
+            else:
+                raise RuntimeError(
+                    "Your metadata store does not have a username. Please "
+                    "provide it either by defining it upon registration or "
+                    "through a MySQL secret."
+                )
+
+        # Set the password
+        if self.password:
+            if secret and secret.password:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a password. Please make sure that you only use one."
+                )
+            else:
+                config.password = self.password
+        else:
+            if secret and secret.user:
+                config.password = self.password
+
+        # Set the SSL configuration if there is one
+        if secret:
+            secret_folder = Path(
+                GlobalConfiguration().config_directory, str(self.uuid)
```

Comment:
This will not pollute the global config directory as much:

```suggestion
                get_global_config_directory(), "mysql-metadata", str(self.uuid)
```

Reply:
Good catch! Changed it!